ethereumj-core/src/main/java/org/ethereum/config/blockchain/AbstractDaoConfig.java

```diff
@@ -0,0 +1,84 @@
+package org.ethereum.config.blockchain;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ethereum.config.BlockchainConfig;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.core.Transaction;
+import org.ethereum.validator.BlockHeaderRule;
+import org.ethereum.validator.BlockHeaderValidator;
+import org.ethereum.validator.ExtraDataPresenceRule;
+import org.spongycastle.util.encoders.Hex;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by Stan Reshetnyk on 26.12.16.
+ */
+public abstract class AbstractDaoConfig extends FrontierConfig {
+
+    /**
+     * Hardcoded values from live network
+     */
+    public static final long ETH_FORK_BLOCK_NUMBER = 1_920_000;
+    // "dao-hard-fork" encoded message
+    public static final byte[] DAO_EXTRA_DATA = Hex.decode("64616f2d686172642d666f726b");
+    private final long EXTRA_DATA_AFFECTS_BLOCKS_NUMBER = 10;
+
+    // MAYBE find a way to remove block number value from blockchain config
+    protected long forkBlockNumber;
+
+    // set in child classes
+    protected boolean supportFork;
+
+    private List<Pair<Long, BlockHeaderValidator>> validator;
+
+    private BlockchainConfig parent;
+
+    protected void initDaoConfig(BlockchainConfig parent, long forkBlockNumber) {
+        this.parent = parent;
+        this.constants = parent.getConstants();
+        this.forkBlockNumber = forkBlockNumber;
+        BlockHeaderRule rule = new ExtraDataPresenceRule(DAO_EXTRA_DATA, supportFork);
+        validator = Arrays.asList(Pair.of(forkBlockNumber, new BlockHeaderValidator(Arrays.asList(rule))));
+    }
+
+    /**
+     * Miners should include marker for initial 10 blocks. Either "dao-hard-fork" or ""
+     */
+    @Override
+    public byte[] getExtraData(byte[] minerExtraData, long blockNumber) {
+        if (blockNumber >= forkBlockNumber && blockNumber < forkBlockNumber + EXTRA_DATA_AFFECTS_BLOCKS_NUMBER ) {
+            if (supportFork) {
+                return DAO_EXTRA_DATA;
+            } else {
+                return new byte[0];
```

Comment:
As far as I understand those who were not supporting DAO fork could just not upgrade their clients and as result those blocks have any data but empty: http://gastracker.io/block/1920000
With this check ETC would not work